sql-plugin/src/main/scala/com/nvidia/spark/rapids/basicPhysicalOperators.scala

```diff
@@ -406,56 +411,81 @@ case class GpuSampleExec(lowerBound: Double, upperBound: Double, withReplacement
     val numOutputRows = gpuLongMetric(NUM_OUTPUT_ROWS)
     val numOutputBatches = gpuLongMetric(NUM_OUTPUT_BATCHES)
     val opTime = gpuLongMetric(OP_TIME)
+
     val rdd = child.executeColumnar()
-    if (withReplacement) {
-      new GpuPartitionwiseSampledRDD(
-        rdd,
-        new GpuPoissonSampler(upperBound - lowerBound, useGapSamplingIfPossible = false,
-          numOutputRows, numOutputBatches, opTime),
-        preservesPartitioning = true,
-        seed)
-    } else {
+    val useGpuToSample = new RapidsConf(conf).isFastSampleEnabled
```

Comment:
nit: I think it would be cleaner if we had a `GpuFastSampleExec` and a `GpuSampleExec`, then we can select which one to use when we replace it.  Part of this is because creating a RapidsConf is not cheap.

Reply:
Yes, done.